src/diagnostic_variables.jl

```diff
@@ -4,19 +4,65 @@ struct Tendencies{NF<:AbstractFloat}
     Tabs_tend    ::Array{Complex{NF},3}      # Absolute temperature [K]
     logp0_tend   ::Array{Complex{NF},2}      # Log of surface pressure [log(Pa)]
     humid_tend   ::Array{Complex{NF},3}      # Specific humidity [g/kg]
+    u_tend       ::Array{Complex{NF},3}      # zonal velocity
+    v_tend       ::Array{Complex{NF},3}      # meridonal velocity
+
 end
 
+
+
+
 struct GridVariables{NF<:AbstractFloat}
-    u       ::Array{NF,3}       # Zonal velocity [m/s]
-    v       ::Array{NF,3}       # Meridional velocity [m/s]
-    Tabs    ::Array{NF,3}       # Absolute temperature [K]
-    logp0   ::Array{NF,2}       # Log of surface pressure [log(Pa)]
-    geopot  ::Array{NF,3}       # 
-    humid   ::Array{NF,3}      # Specific humidity [g/kg]
+   # u       ::Array{NF,3}       # Zonal velocity [m/s]
+    #v       ::Array{NF,3}       # Meridional velocity [m/s]
+   # Tabs    ::Array{NF,3}       # Absolute temperature [K]
+   # logp0   ::Array{NF,2}       # Log of surface pressure [log(Pa)]
+   # geopot  ::Array{NF,3}       # 
+   # humid   ::Array{NF,3}      # Specific humidity [g/kg]
+
+    vor_grid    ::Array{NF,3}  # Gridpoint field of vorticity
+    div_grid    ::Array{NF,3}  # Gridpoint field of divergence
+    Tabs_grid   ::Array{NF,3}  # Gridpoint field of absolute temperature [K]
+    logp0_grid  :: Array{NF,2} # Gridpoint field of surface pressure logarithm [log(Pa)]
+    humid_grid  ::Array{NF,3}  # Gridpoint field of specific_humidity
+    geopot_grid ::Array{NF,3}  # Gridpoint field of geopotential
+    tr_grid     ::Array{NF,3}  # Gridpoint field of tracers
+    u_grid      ::Array{NF,3}  # Gridpoint field of zonal velocity
+    v_grid      ::Array{NF,3}  # Gridpoint field of meridional velocity
+
+
+    Tabs_grid_anomaly   ::Array{NF,3}  # Gridpoint field of absolute temperature anomaly [K]
+
+    
 end
 
+
+struct MiscellaneousVariables{NF<:AbstractFloat}
```

Comment:
I think once we know better what all else is needed we can rename this to be more specific?